### PR TITLE
Sort Pods by workload logic for ImagePullJob pre-downloading

### PR DESCRIPTION
**Pull request by FillZpp**
Signed-off-by: FillZpp <FillZpp.pub@gmail.com>

### Ⅰ. Describe what this PR does
Sort Pods by workload logic for ImagePullJob pre-downloading

Reference to proposal [image pre-download during in-place update](https://github.com/openkruise/kruise/pull/570).

### CODE REVIEW

File: pkg/control/sorting/pod_sorting.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SortPods sorts the given Pods according the owner workload logic.
+func SortPods(reader client.Reader, ns string, owner metav1.OwnerReference, pods []*v1.Pod) ([]*v1.Pod, error) {
```
Comment (furykerry) [inline pkg/control/sorting/pod_sorting.go:39]: pls comments the usage of the sorting, only for image predownload function?

File: pkg/util/nodeimages/nodeimages_util.go [hunk 2]
```diff
@@ -79,10 +81,28 @@ func GetNodeImagesForJob(reader client.Reader, job *appsv1alpha1.ImagePullJob) (
 			return nil, err
 		}
 
-		nodeImageNames := sets.NewString()
+		pods := make([]*v1.Pod, 0, len(podList.Items))
 		for i := range podList.Items {
 			pod := &podList.Items[i]
-			if !kubecontroller.IsPodActive(pod) || pod.Spec.NodeName == "" || nodeImageNames.Has(pod.Spec.NodeName) {
+			if !kubecontroller.IsPodActive(pod) || pod.Spec.NodeName == "" {
```
Comment (furykerry) [inline pkg/util/nodeimages/nodeimages_util.go:87]: why drop the nodeImage check? sorting with less pods  can be more efficient

**Review by furykerry — CHANGES_REQUESTED** [review]

File: pkg/control/sorting/pod_sorting.go (hunk 1 above)
Comment (FillZpp) [reply pkg/control/sorting/pod_sorting.go:40]: done

File: pkg/util/nodeimages/nodeimages_util.go (hunk 2 above)
Comment (FillZpp) [reply pkg/util/nodeimages/nodeimages_util.go:87]: we can not distinct pods with same node here, it may cause incorrect pre-download sequence

**Review by furykerry — APPROVED** [review]

**Pull request merged**